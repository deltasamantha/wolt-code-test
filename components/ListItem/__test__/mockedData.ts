import {Slot} from "../../../models/DisplayTimeSlot";

export const slotsData: Slot[] = [
  {
    open: 36000,
    close: 43200,
  },
  {
    open: 43500,
    close: 75600,
  },
];
