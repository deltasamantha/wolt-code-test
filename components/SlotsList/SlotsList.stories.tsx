import {Meta, StoryObj} from "@storybook/react-native";
import {View} from "react-native";

import SlotsList, {Props} from "./SlotsList";
import {storyStyles} from "./SlotsList.styles";
import {Slot} from "../../models/TimeSlot";

const mockedSlots: Slot[] = [
  {
    open: 36000,
    close: 43200,
  },
  {
    open: 43500,
    close: 75600,
  },
];

const ListItemMeta: Meta<Props> = {
  title: "SlotsList",
  component: SlotsList,
  argTypes: {
    slots: {
      description: "Available slots for the day",
      control: "object",
    },
  },
  decorators: [
    Story => (
      <View style={storyStyles.rootContainer}>
        <Story />
      </View>
    ),
  ],
};

export default ListItemMeta;

export const Primary: StoryObj<Props> = {
  args: {
    slots: [],
  },
};

export const ItemWithSingleSlot: StoryObj<Props> = {
  args: {
    slots: mockedSlots.slice(1),
  },
};

export const ItemWithMultipleSlot: StoryObj<Props> = {
  args: {
    slots: mockedSlots,
  },
};
