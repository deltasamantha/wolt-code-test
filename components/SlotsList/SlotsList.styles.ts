import {StyleSheet} from "react-native";

import theme from "../../theme/theme";

export const styles = StyleSheet.create({
  rootContainer: {
    flexDirection: "column",
    alignItems: "flex-end",
  },
});

export const storyStyles = StyleSheet.create({
  rootContainer: {
    flex: 1,
    paddingHorizontal: theme.spacing.lg,
    backgroundColor: theme.colors.colorBackground,
    justifyContent: "center",
  },
});
