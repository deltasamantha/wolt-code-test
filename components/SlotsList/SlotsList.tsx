import React from "react";
import {View} from "react-native";

import {styles} from "./SlotsList.styles";
import locale from "../../config/Locale";
import {Slot} from "../../models/TimeSlot";
import {getTimeSlotTimeStamp} from "../../utils/DateTimeUtils";
import AppText from "../AppText/AppText";

export interface Props {
  slots: Slot[];
}

const SlotsList: React.FC<Props> = ({slots}: Props) => {
  return (
    <View style={styles.rootContainer}>
      {!slots.length && (
        <AppText type="regular" color="light" size="m">
          {locale.closed}
        </AppText>
      )}
      {!!slots &&
        slots.map((slot, index) => (
          <AppText type="regular" color="dark" size="m" key={index}>
            {getTimeSlotTimeStamp(slot.open, slot.close)}
          </AppText>
        ))}
    </View>
  );
};

export default SlotsList;
