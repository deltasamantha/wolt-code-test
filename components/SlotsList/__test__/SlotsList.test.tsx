import {render, cleanup, screen} from "@testing-library/react-native";

import {slotsData} from "./mockedData";
import locale from "../../../config/Locale";
import SlotsList from "../SlotsList";

afterEach(cleanup);

describe("SlotsList", () => {
  it("Should render SlotsList properly", () => {
    const {toJSON} = render(<SlotsList slots={[]} />);
    expect(toJSON()).toMatchSnapshot();
  });

  it("Should render placeholder if no slots available", () => {
    const placeholder = locale.closed;
    render(<SlotsList slots={[]} />);
    const placeholderText = screen.getByText(placeholder);
    expect(placeholderText).toBeTruthy();
  });

  it("Should render slots", () => {
    render(<SlotsList slots={slotsData} />);
    const slots = screen.getAllByText("PM", {exact: false});
    expect(slots).toBeTruthy();
    expect(slots.length).toBe(2);
  });
});
