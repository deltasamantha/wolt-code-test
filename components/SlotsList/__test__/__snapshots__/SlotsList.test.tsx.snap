// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SlotsList Should render SlotsList properly 1`] = `
<View
  style={
    {
      "alignItems": "flex-end",
      "flexDirection": "column",
    }
  }
>
  <Text
    style={
      [
        {
          "color": "#A1A2A4",
          "fontFamily": "Roboto_400Regular",
          "fontSize": 16,
          "lineHeight": 22,
        },
        undefined,
      ]
    }
  >
    Closed
  </Text>
</View>
`;
